r-package/R/read_municipality.R

```diff
@@ -1,139 +1,139 @@
-          x <- code_muni
-          temp_sf <- subset(temp_sf, substr(code_muni,1,2)==x)
-          return(temp_sf)}
-
-        else if (is.character(code_muni)){
-          x <- code_muni
-          temp_sf <- subset(temp_sf, substr(abbrev_state,1,2)==x)
-          return(temp_sf)}
-        }
-
-
-  # if input is a muni_code
-      else if(nchar(code_muni)==7) {
-
-    # invalid muni_code
-
-      if( !( code_muni %in% temp_sf$code_muni)){
-        stop("Error: Invalid value to argument code_muni")}
-
-    # valid muni_code
-        else {
-            x <- code_muni
-            temp_sf <- subset(temp_sf, code_muni==x)
-            return(temp_sf)}
-      }
-
-      else if(nchar(code_muni)!=7 | nchar(code_muni)!=2) {
-        stop("Error: Invalid value to argument code_muni")}
-
-        } else {
-
-
-# BLOCK 2.2 From 2000 onwards  ----------------------------
-
-# 2.2 Verify code_muni Input
-
-  # if code_muni=="all", read the entire country
-    if(code_muni=="all"){ message("Loading data for the whole country. This might take a few minutes.\n")
-
-      # list paths of files to download
-      file_url <- as.character(temp_meta$download_path)
-
-      # download files
-      temp_sf <- download_gpkg(file_url, progress_bar = showProgress)
-      return(temp_sf)
-    }
-
-  else if( !(substr(x = code_muni, 1, 2) %in% temp_meta$code) & !(substr(x = code_muni, 1, 2) %in% temp_meta$code_abrev)){
-
-      stop("Error: Invalid Value to argument code_muni.")
-
-  } else{
-
-    # list paths of files to download
-    if (is.numeric(code_muni)){ file_url <- as.character(subset(temp_meta, code==substr(code_muni, 1, 2))$download_path) }
-    if (is.character(code_muni)){ file_url <- as.character(subset(temp_meta, code_abrev==substr(code_muni, 1, 2))$download_path) }
-
-    # download files
-    sf <- download_gpkg(file_url, progress_bar = showProgress)
-
-    # input is a state code
-    if(nchar(code_muni)==2){
-      sf <- subset(sf, code_state==substr(code_muni, 1, 2))
-        return(sf) }
-
-    # input is a municipality code
-    else if(code_muni %in% sf$code_muni){
-          x <- code_muni
-          sf <- subset(sf, code_muni==x)
-          return(sf)
-      } else{
-          stop("Error: Invalid Value to argument code_muni.")
-      }
-    }
-    }
-  }
-
+#' Download shape files of Brazilian municipalities as sf objects
+#'
+#' Data at scale 1:250,000, using Geodetic reference system "SIRGAS2000" and CRS(4674).
+#'
+#' @param year Year of the data (defaults to 2010).
+#' @param code_muni The 7-digit identification code of a municipality. If `code_muni = "all"`
+#' (default), all municipalities of the country will be downloaded. Alternativelly, if the two-digit
+#' identification code or a two-letter uppercase abbreviation of a state is passed, e.g. `33` or
+#' `"RJ"`, all municipalities of that state will be downloaded. Municipality identification codes
+#' are defined in \url{https://www.ibge.gov.br/explica/codigos-dos-municipios.php}.
```

Comment:
The link seems to be broken

Reply:
This is curious. It was working when I included it. I will try to find a solution ASAP.

Reply:
The website seems to be under maintenance -- see https://cidades.ibge.gov.br/.